code/dataobjects/Block.php

```diff
@@ -428,6 +428,14 @@ public function CSSClasses($stopAtClass = 'DataObject')
     }
 
     /**
+     * Access current page scope from Block templates with $CurrentPage
+     * @return SiteTree | Controller
+     */
+    public function getCurrentPage() {
+        return Director::get_current_page();
```

Comment:
`Director::get_current_page()` only gives you the base SiteTree/Page object, not the controller so it's not as useful actually. You wouldn't be able to use any page controller methods with what I've done here.

Probably is a good idea to use `Controller::curr()` instead... `getCurrentPage` doesn't quite seem semantic as a name choice in that case but `$CurrentPage` seems more intuitive to me to use in templates than `$CurrentController` so maybe it's okay?